Add Variant Allele Frequency in Plots tab - GSoC'25

rishisulakhe opened the pull request:
Describe changes proposed in this pull request:
- Added variant allele frequency in plots tab

## Any screenshots or GIFs?
![Screenshot 2025-06-07 003357](https://github.com/user-attachments/assets/894252a9-348f-48e8-96a6-2a688d20a037)
![Screenshot 2025-06-07 003517](https://github.com/user-attachments/assets/42006943-bfde-4cf3-b863-6ef63e613f77)

## Notify reviewers
@onursumer

onursumer commented on src/shared/components/plots/PlotsTabUtils.tsx:
```diff
@@ -1146,101 +1146,137 @@ export function makeAxisDataPromise_Molecular_MakeMutationData(
+                    value = isProfiled ? 0 : NaN;
+                } else {
+                    const vafs = sampleMutations
+                        .map(m => {
+                            if (
+                                Number.isInteger(m.tumorAltCount) &&
+                                Number.isInteger(m.tumorRefCount)
+                            ) {
+                                const total = m.tumorAltCount + m.tumorRefCount;
+                                return total > 0
+                                    ? m.tumorAltCount / total
+                                    : null;
+                            }
+                            return null;
```
Let's reuse an existing utility function for this if possible.

https://github.com/cBioPortal/cbioportal-frontend/blob/2894692dc0ae1e02b7985aa2fe49fb44f9c7631d/src/shared/lib/MutationUtils.tsx#L382-L399

onursumer commented on src/shared/components/plots/PlotsTab.tsx:
```diff
@@ -220,6 +220,7 @@ export enum MutationCountBy {
     MutationType = 'MutationType',
     MutatedVsWildType = 'MutatedVsWildType',
     DriverVsVUS = 'DriverVsVUS',
+    VAF = 'VAF',
```
Should we just spell it out for clarity?

```suggestion
    VariantAlleleFrequency = 'VariantAlleleFrequency',
```

onursumer reviewed (changes requested):
It looks good in general, thanks! Just requested changes for a few minor things.

Also, looks like `prettier` is failing. You might have forgotten to run the commit hooks.

rishisulakhe commented on src/shared/components/plots/PlotsTabUtils.tsx:
```diff
@@ -1146,101 +1147,128 @@ export function makeAxisDataPromise_Molecular_MakeMutationData(
+                        const report = getVariantAlleleFrequency(m);
+                        if (report !== null) {
+                            vafs.push(report.vaf);
+                        }
```
Sure, will use these functions.

onursumer commented:
Just noticed that certain label name changes are not desirable. 

Examples:

`Mutations` -> `Mutations Type`

before:
<img width="210" height="41" alt="TP53-Mutations" src="https://github.com/user-attachments/assets/4e940a79-c145-461e-a34f-8876692cbcc9" />

after:
<img width="160" height="44" alt="mutations_type" src="https://github.com/user-attachments/assets/b400b44c-30dd-4fbf-bd46-6b78df17f6c0" />

`TP53: Mutations (MSK-IMPACT)` -> `TP53: Mutations Type`

before:
<img width="217" height="40" alt="Screenshot 2025-09-22 at 5 46 06 PM" src="https://github.com/user-attachments/assets/0f6c1107-819a-4a30-8ab1-79acc8743904" />

after:
<img width="186" height="57" alt="Screenshot 2025-09-22 at 5 45 34 PM" src="https://github.com/user-attachments/assets/69c9d6ce-9b7b-4afe-af5f-ac978e98fc24" />

rishisulakhe commented:
Hi @inodb
Thanks for the feedback! I've updated the implementation to filter out samples without mutations in that specific gene and samples with mutations but no vaf data. Also updated the top status bar.

<img width="1591" height="800" alt="Screenshot 2025-10-03 203123" src="https://github.com/user-attachments/assets/24d6032c-3e86-46d3-bf5c-e5d32f034ecc" />
